Kuzzle/API/Controllers/AdminController.cs

```diff
@@ -0,0 +1,162 @@
+﻿using System;
+using System.Threading.Tasks;
+using KuzzleSdk.Utils;
+using Newtonsoft.Json.Linq;
+
+namespace KuzzleSdk.API.Controllers {
+  public class AdminController : BaseController {
+    internal AdminController(IKuzzleApi k) : base(k) { }
+
+
+    /// <summary>
+    /// Asynchronously create a snapshot of Kuzzle's state.
+    /// Depending on the configuration of Kuzzle.
+    /// </summary>
+    public async Task<bool> DumpAsync() {
+
+      Response response = await api.QueryAsync(new JObject {
+        {"controller", "admin"},
+        {"action", "dump"}
+      });
+
+      return (bool)response.Result["acknowledge"];
+    }
+
+    /// <summary>
+    /// Load fixtures into the storage layer.
+    /// </summary>
+    public async Task<bool> LoadFixturesAsync(JObject indexName, bool waitForRefresh = false) {
+
+      JObject query = new JObject {
+        {"controller", "admin"},
+        {"action", "loadFixtures"},
+        {"body", new JObject {
+            {"index-name", indexName}
+          }
+        }
+      };
+
+      QueryUtils.HandleRefreshOption(query, waitForRefresh);
+
+      Response response = await api.QueryAsync(query);
+
+      return (bool)response.Result["acknowledge"];
+    }
+
+    /// <summary>
+    /// Apply mappings to the storage layer.
+    /// </summary>
+    public async Task<bool> LoadMappingsAsync(JObject indexName, bool waitForRefresh = false) {
+
+      JObject query = new JObject {
+        {"controller", "admin"},
+        {"action", "loadMappings"},
+        {"body", new JObject {
+            {"index-name", indexName}
+          }
+        }
+      };
+
+      QueryUtils.HandleRefreshOption(query, waitForRefresh);
+
+      Response response = await api.QueryAsync(query);
+
+      return (bool)response.Result["acknowledge"];
+    }
+
+    /// <summary>
+    /// Load roles, profiles and users into the storage layer.
+    /// </summary>
+    public async Task<bool> LoadSecuritiesAsync(JObject body, bool waitForRefresh = false) {
```

Comment:
You couldn't know that obviously, but routes returning the following kind of results: `{acknowledge: true}` will never have a `false` value. What it says is that the query is correctly formed and is thus accepted. 
If something's wrong, the `result` part of the response will be null and the error part will be filled with the appropriate information.

So all methods wrapping API routes returning that kind of response must return a `Task` instead of a `Task<bool>`

Reply:
I didn't know that thanks.
This has been changed to return `Task` instead of `Task<bool>`.